src/test/lib/storage/index/partial_hash/partial_hash_index_test.cpp

```diff
@@ -175,26 +198,41 @@ TEST_F(PartialHashIndexTest, Add) {
   EXPECT_TRUE(index->get_indexed_chunk_ids().contains(ChunkID{1}));
   EXPECT_TRUE(index->get_indexed_chunk_ids().contains(ChunkID{2}));
 
-  EXPECT_EQ(std::distance(index->cbegin(), index->cend()), 21);
-  EXPECT_EQ(std::distance(index->null_cbegin(), index->null_cend()), 3);
-  EXPECT_EQ(*index->range_equals("new1").first, (RowID{ChunkID{2}, ChunkOffset{0}}));
+  EXPECT_EQ(std::distance(index_impl->cbegin(), index_impl->cend()), 21);
+  EXPECT_EQ(std::distance(index_impl->null_cbegin(), index_impl->null_cend()), 3);
+  EXPECT_EQ(*index_impl->range_equals("new1").first, (RowID{ChunkID{2}, ChunkOffset{0}}));
 
   EXPECT_EQ(index->insert_entries(chunks_to_add), 0);
 }
 
 TEST_F(PartialHashIndexTest, InsertIntoEmpty) {
-  auto empty_index = PartialHashIndex(DataType::String, ColumnID{0});
-
   auto chunks_to_add =
       std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>>{std::make_pair(ChunkID{0}, table->get_chunk(ChunkID{0}))};
-  EXPECT_EQ(empty_index.insert_entries(chunks_to_add), 1);
+  EXPECT_EQ(empty_index->insert_entries(chunks_to_add), 1);
 
-  EXPECT_EQ(empty_index.get_indexed_chunk_ids().size(), 1);
-  EXPECT_TRUE(empty_index.get_indexed_chunk_ids().contains(ChunkID{0}));
+  EXPECT_EQ(empty_index->get_indexed_chunk_ids().size(), 1);
+  EXPECT_TRUE(empty_index->get_indexed_chunk_ids().contains(ChunkID{0}));
 
-  EXPECT_EQ(std::distance(empty_index.cbegin(), empty_index.cend()), 7);
-  EXPECT_EQ(std::distance(empty_index.null_cbegin(), empty_index.null_cend()), 1);
-  EXPECT_EQ(*empty_index.range_equals("hotel").first, (RowID{ChunkID{0}, ChunkOffset{0}}));
+  EXPECT_EQ(std::distance(empty_index_impl->cbegin(), empty_index_impl->cend()), 7);
+  EXPECT_EQ(std::distance(empty_index_impl->null_cbegin(), empty_index_impl->null_cend()), 1);
+  EXPECT_EQ(*empty_index_impl->range_equals("hotel").first, (RowID{ChunkID{0}, ChunkOffset{0}}));
+}
+
+TEST_F(PartialHashIndexTest, Remove) {
+  EXPECT_EQ(index->remove_entries(std::vector<ChunkID>{ChunkID{0}}), 1);
+
+  EXPECT_EQ(index->get_indexed_chunk_ids(), (std::unordered_set<ChunkID>{ChunkID{1}}));
+  EXPECT_EQ(std::distance(index_impl->cbegin(), index_impl->cend()), 7);
+  EXPECT_EQ(std::distance(index_impl->null_cbegin(), index_impl->null_cend()), 1);
+  EXPECT_EQ(index_impl->range_equals("hotel").first, index_impl->cend());
+
+  EXPECT_EQ(index->remove_entries(std::vector<ChunkID>{ChunkID{0}}), 0);
+}
+
+TEST_F(PartialHashIndexTest, RemoveFromEmpty) {
+  EXPECT_EQ(empty_index->remove_entries(std::vector<ChunkID>{ChunkID{0}}), 0);
+  EXPECT_EQ(empty_index->get_indexed_chunk_ids().size(), 0);
+  EXPECT_EQ(empty_index_impl->cbegin(), empty_index_impl->cend());
 }
 
```

Comment:
Please add test cases for `range_equals_with_iterators` and `range_not_equals_with_iterators`.